src/java/org/apache/cassandra/db/MultiClusteringBuilder.java

```diff
@@ -0,0 +1,437 @@
+
+        final List<ByteBuffer> values;
+        final Kind kind;
+
+
+        private ClusteringElements(List<ByteBuffer> values, Kind kind)
+        {
+            this.values = values;
+            this.kind = kind;
+        }
+
+        public static ClusteringElements point(ByteBuffer value)
+        {
+            return point(Collections.singletonList(value));
+        }
+
+        public static ClusteringElements point(List<ByteBuffer> values)
+        {
+            return new ClusteringElements(values, Kind.POINT);
+        }
+
+        public static ClusteringElements bound(ByteBuffer value, Bound bound, boolean inclusive)
+        {
+            return bound(Collections.singletonList(value), bound, inclusive);
+        }
+
+        public static ClusteringElements bound(List<ByteBuffer> values, Bound bound, boolean inclusive)
+        {
+            Kind kind = bound.isStart()
+                    ? (inclusive ? Kind.INCL_START : Kind.EXCL_START)
+                    : (inclusive ? Kind.INCL_END : Kind.EXCL_END);
+            return new ClusteringElements(values, kind);
+        }
+
+        public  boolean isBound()
+        {
+            return kind != Kind.POINT;
+        }
+
+        public boolean isStart()
+        {
+            return kind == ClusteringElements.Kind.EXCL_START ||
+                   kind == ClusteringElements.Kind.INCL_START;
+        }
+
+        public boolean isInclusive()
+        {
+            return kind == Kind.INCL_START ||
+                   kind == Kind.INCL_END ||
+                   kind == Kind.POINT;
+        }
+
+        public String toString()
+        {
+            return "Element{" +
+                   "kind=" + kind +
+                   ", value=" + values +
+                   '}';
+        }
+    }
+
+    /**
+     * The table comparator.
+     */
+    private final ClusteringComparator comparator;
+
+    /**
+     * Columns corresponding to the already added elements.
+     */
+    private final List<ColumnMetadata> columns = new ArrayList<>();
+
+    /**
+     * The elements of the clusterings.
+     */
+    private List<ClusteringElements> clusterings = Collections.singletonList(ClusteringElements.ROOT);
+
+
+    /**
+     * <code>true</code> if the clusterings have been build, <code>false</code> otherwise.
+     */
+    private boolean built;
+
+    /**
+     * <code>true</code> if the clusterings contains some <code>null</code> elements.
+     */
+    private boolean containsNull;
+
+    /**
+     * <code>true</code> if the composites contains some <code>unset</code> elements.
+     */
+    private boolean containsUnset;
+
+    /**
+     * <code>true</code> if the composites contains some slice bound elements.
+     */
+    private boolean containsSliceBound;
+
+
+    private MultiClusteringBuilder(ClusteringComparator comparator)
+    {
+        this.comparator = comparator;
+    }
+
+    /**
+     * Creates a new empty {@code MultiCBuilder}.
```

Comment:
```suggestion
     * Creates a new empty {@code MultiClusteringBuilder}.
```

Reply:
I think this nit hasn't actually been resolved, it still says `{@code MultiCBuilder}`.